superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/FiltersConfigForm.tsx

```diff
@@ -115,16 +122,33 @@ export const FiltersConfigForm: React.FC<FiltersConfigFormProps> = ({
   const hasColumn =
     hasDataset && !FILTERS_WITHOUT_COLUMN.includes(formFilter?.filterType);
 
+  const datasetId = formFilter?.dataset?.value;
+
+  useEffect(() => {
+    if (datasetId) {
```

Comment:
If https://github.com/apache/superset/pull/14313 is merged first, that won't be necessary - metrics can be accessed in `formFilter.dataset?.metrics`

Reply:
Oh nice, let's get that one in first then 👍